Makefile

```
@@ -110,3 +110,7 @@ markdownlint:
 
 markdownlint-fix:
 	docker run -v $(CURDIR):/workdir --rm  ghcr.io/igorshubovych/markdownlint-cli:latest  "**/*.md" --config "/workdir/docs/markdownlint-rules.yaml" --fix --ignore "/workdir/CHANGELOG.md"
+
+yamllint:
+	@docker run --rm $$(tty -s && echo "-it" || echo) -v $(PWD):/data cytopia/yamllint:latest .github docs
```

```suggestion
	@docker run --rm -t -v $(PWD):/data cytopia/yamllint:latest .github docs
```

i like this `$$(tty -s && echo "-it" || echo) ` but i do think `-t` should be enough - we are not providing any input. but if we want to stick using your magic TTY logic, we should extract it into a variable and apply it to all of the targets here :)

---

Yes! this is working with the `-t` flag too. I'll change it. 
I didn't get you with how we can include the directories inside the yamllint configuration file. we have an option to [ignore](https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths) the files but I can't seem to find any way to include the files.
There's a default configuration of the [yaml-files](https://yamllint.readthedocs.io/en/stable/configuration.html#default-configuration). I tried to add `.github/**` and `docs/**` in this and remove `.github` and `docs` from the make target but it is not working.

---

you do have multiple options - https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths i would opt in for a `.yamlignore` file and the `ignore-from-file: [.gitignore, .yamlignore]` directive - but there are other options and you can explore :)

---

so should i ignore all the files except `docs` and `.github`?

---

@mowies can you guide us here?

---

you should be able to set explicit includes with the `yaml-files:` flag in the `.yamllint` config file.
so you could add there:
```
yaml-files:
- ".github/**/*.yaml"
- ".github/**/*.yml"
- "docs/**/*.yaml"
- "docs/**/*.yml"
```

The docs for this can be found [here](https://yamllint.readthedocs.io/en/stable/configuration.html#yaml-files-extensions)

---

@mowies I did try this earlier and just tried again but it is not working. 
Make command- `@docker run --rm -t -v $(PWD):/data cytopia/yamllint:$(YAMLLINT_VERSION) .`

yaml-files command- 
```
yaml-files:
- ".github/**/*.yaml"
- ".github/**/*.yml"
- "docs/**/*.yaml"
- "docs/**/*.yml"
```

these are the commands i am using right now in the `Makefile` and the `.yamllint`. When I am trying to run `make yamllint` nothing shows up.

---

I created an issue for this in the yamllint repo: https://github.com/adrienverge/yamllint/issues/546

---

Let's for now go with just defining `.github docs` in the yamllint command in the makefile. That works for now until we have a solution for my created issue I would say.

---

Sure! we can discuss about the rules until then.